src/addrman.h

```diff
@@ -483,15 +486,28 @@ friend class CAddrManTest;
 
         // Prune new entries with refcount 0 (as a result of collisions).
         int nLostUnk = 0;
-        for (std::map<int, CAddrInfo>::const_iterator it = mapInfo.begin(); it != mapInfo.end(); ) {
+#if __cplusplus >= 201402L // C++14 or higher, erasing from unordered_map is guaranteed to preserve the order of not deleted elements
+        for (MapInfo::const_iterator it = mapInfo.begin(); it != mapInfo.end(); ) {
             if (it->second.fInTried == false && it->second.nRefCount == 0) {
-                std::map<int, CAddrInfo>::const_iterator itCopy = it++;
+                auto itCopy = it++;
                 Delete(itCopy->first);
                 nLostUnk++;
             } else {
-                it++;
+                ++it;
             }
         }
+#else // pre C++14, drop this once we switch to C++14 or higher
+        std::vector<MapInfo::const_iterator> toDelete;
```

Comment:
Nit: new code should use the coding style described in doc/developer-.nodes.md (e.g. `to_delete` instead).

Reply:
Done